src/app/shared/model/story.model.ts

```diff
@@ -134,9 +135,20 @@ export class StoryModel extends BaseModel {
     }
   }
 
-  get unsavedImage(): ImageModel {
-    let img = new ImageModel(this.parent, this.doc, null);
-    return img.isStored() && !img.isDestroy ? img : null;
+  addImage(upload: Upload): ImageModel {
+    let image = new ImageModel(this.doc, upload);
+    this.images.push(image);
```

Comment:
Ah, good call.  I saw a change detection bug coming from there at one point, but assumed it was a fluke!

Will try making a new array.  Which is probably a good policy anytime we're adding/removing a model from the array.